common/lib/xmodule/xmodule/contentstore/mongo.py

```diff
@@ -431,18 +431,28 @@ def copy_all_course_assets(self, source_course_key, dest_course_key):
                 asset_id = six.text_type(
                     dest_course_key.make_asset_key(asset_key['category'], asset_key['name']).for_branch(None)
                 )
-
-            self.fs.put(
-                source_content.read(),
-                _id=asset_id, filename=asset['filename'], content_type=asset['contentType'],
-                displayname=asset['displayname'], content_son=asset_key,
-                # thumbnail is not technically correct but will be functionally correct as the code
-                # only looks at the name which is not course relative.
-                thumbnail_location=asset['thumbnail_location'],
-                import_path=asset['import_path'],
-                # getattr b/c caching may mean some pickled instances don't have attr
-                locked=asset.get('locked', False)
-            )
+            try:
+                self.fs.put(
+                    source_content.read(),
+                    _id=asset_id, filename=asset['filename'], content_type=asset['contentType'],
+                    displayname=asset['displayname'], content_son=asset_key,
+                    # thumbnail is not technically correct but will be functionally correct as the code
+                    # only looks at the name which is not course relative.
+                    thumbnail_location=asset['thumbnail_location'],
+                    import_path=asset['import_path'],
+                    # getattr b/c caching may mean some pickled instances don't have attr
+                    locked=asset.get('locked', False)
+                )
+            except FileExists:
+                self.fs.delete(_id=asset_id)
+                self.fs.put(
+                    source_content.read(),
+                    _id=asset_id, filename=asset['filename'], content_type=asset['contentType'],
+                    displayname=asset['displayname'], content_son=asset_key,
+                    thumbnail_location=asset['thumbnail_location'],
+                    import_path=asset['import_path'],
+                    locked=asset.get('locked', False)
+                )
```

Comment:
1. Move duplicate code in function `self.fs.put(....)` 
2. Unit test for this change?

Reply:
Adding tests

Reply:
Can you try and check if there is a method to check `self.fs.has_asset`?

Reply:
@awaisdar001 there is a method to check if the asset exists. But I think there is no need to check every asset as that would be an overhead. We are just dealing with an edge case so handling the FileExists seems a better option to me

Reply:
Agree. !!

Reply:
+1 for @AhtishamShahid comment above.